src/cache/index.js

```
@@ -1,103 +1,133 @@
       // because setex does not support setting objects.
       const stringifiedData = JSON.stringify(data);
       await client.setex(key, ttl || cacheDuration, stringifiedData);
     } catch (error) {
-      logger.error(error, `Failed to store a cache item with key '${key}'`, this.configuration);
-      if (this.redisClient && this.redisClient.master) {
-        this.redisClient.master.checkStatus();
-      }
+      logger.error();
+      logger.error(`redis:primary Cannot SETEX from ${key} as an error occurred: ${error.message}`);
+      logger.error();
     }
   }
 
   async _redisGet(key) {
-    if (!this.redisClient) {
-      throw new Error('Redis client undefined');
-    }
+    const { client, status, ready } = this.getClientAndStatus('primary');
 
-    if (!(this.redisClient.slave.isHealthy && this.redisClient.slave.redis)) {
-      throw new Error('Redis client not ready/healthy');
+    if (!ready) {
+      const message = `redis:reader Cannot GET from ${key} as client is in status ${status}`;
+      logger.warn(message);
+      throw new Error(message);
     }
 
-    const client = this.redisClient.slave.redis;
     try {
       let result;
+
       if (this.configuration.redisGetTimeout) {
         result = await Promise.race([timeout(this.configuration.redisGetTimeout), client.get(key)]);
-        // unstringify the data
-        result = JSON.parse(result);
-        return result;
+      } else {
+        result = await client.get(key);
       }
-      result = await client.get(key);
+
       // unstringify the data
       result = JSON.parse(result);
       return result;
     } catch (error) {
-      logger.error('Failed to get item from cache', error);
-      logger.trace(error);
-      this.redisClient.slave.checkStatus();
+      const message = `redis:reader Cannot GET from ${key} as an error occurred: ${error.message}`;
+      logger.error(message);
+      throw new Error(message);
```

why are we throwing the error now?

---

There is a `try...catch` block in `get` that uses `_redisGet` internally, so I thought it makes more sense to throw an exception rather than returning `null`.

---

It also simplifies the code because there is no need to run an `if(result)` check inside `get()`.

---

yup, fair enough. Do we not need to handle that error from somewhere though? Like, we don't want the API to blow up if the key is not in Redis, right?

---

It's caught as we discussed today.